### PR TITLE
Migrate PermanentDeleteByTeam method from ChannelStore to return erro…

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

This PR migrates `PermanentDeleteByTeam` method.

Fixes https://github.com/mattermost/mattermost-server/issues/14715

Jira ticket: https://mattermost.atlassian.net/browse/MM-25690

### CODE REVIEW

**Comment by rvillablanca** [comment]
We need tracking there so a ticket is required @agnivade :+1: thanks

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -819,10 +819,10 @@ func (s SqlChannelStore) setDeleteAtT(transaction *gorp.Transaction, channelId s
...
 
 // PermanentDeleteByTeam removes all channels for the given team from the database.
-func (s SqlChannelStore) PermanentDeleteByTeam(teamId string) *model.AppError {
+func (s SqlChannelStore) PermanentDeleteByTeam(teamId string) error {
 	transaction, err := s.GetMaster().Begin()
 	if err != nil {
-		return model.NewAppError("SqlChannelStore.PermanentDeleteByTeam", "store.sql_channel.permanent_delete_by_team.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "begin_transaction")
 	}
 	defer finalizeTransaction(transaction)
```
Comment (agnivade) [inline store/sqlstore/channel_store.go:828]: Same again. Wrap `permanentDeleteByTeamT` :)

Comment (agnivade) [reply store/sqlstore/channel_store.go:835]: Perhaps I am missing something, but I don't see the wrapping done here.

**Comment by rvillablanca** [comment]
Conflicts resolved!

**Pull request merged**